src/Stratis.Bitcoin/Controllers/RestApiClientBase.cs

```diff
@@ -118,9 +118,103 @@ public RestApiClientBase(ILoggerFactory loggerFactory, IFederationGatewaySetting
             return null;
         }
 
+        public async Task<Response> SendGetRequestAsync<Response>(string apiMethodName, string arguments = null,
+            CancellationToken cancellation = default(CancellationToken)) where Response : class
+        {
+            HttpResponseMessage response = await this.SendGetRequestAsync(apiMethodName, arguments, cancellation).ConfigureAwait(false);
+
+            // Parse response.
+            if ((response != null) && response.IsSuccessStatusCode && (response.Content != null))
```

Comment:
@noescape00 Can you perhaps on each of these trace log what was false? I have found that with these Http calls its best to exactly why the request failed.

```
if (response == null)
{
    this.logger.LogTrace("(-)[RESPONSE_NULL]");
    return null;
}
```

etc

Reply:
Good point. Fixed